src/main/scala/dx/translator/cwl/CallableTranslator.scala

```diff
@@ -0,0 +1,72 @@
+package dx.translator.cwl
+
+import java.nio.file.Path
+
+import dx.api.DxApi
+import dx.core.ir.{Application, Callable, CallableAttribute, ExecutableKindApplet, Parameter, ParameterAttribute, RuntimeRequirement, Value}
+import dx.core.languages.cwl.CwlUtils
+import dx.translator.{ReorgSettings, RunSpec}
+import dx.cwl.{BooleanValue, CommandInputParameter, CommandLineTool, CommandOutputParameter, DockerRequirement, LongValue, Process, SchemaDefRequirement, ToolTimeLimitRequirement, WorkReuseRequirement, Workflow}
+import dx.translator.RunSpec.{ContainerImage, IgnoreReuseRequirement, TimeoutRequirement}
+
+
+case class CallableTranslator(cwlBundle: CwlBundle,
+                              locked: Boolean,
+                              defaultRuntimeAttrs: Map[String, Value],
+                              reorgAttrs: ReorgSettings,
+                              dxApi: DxApi = DxApi.get) {
+
+
+  def translateCallable(callable: Process, sourceFile: Path): Callable = {
+    callable match {
+      case tool: CommandLineTool =>
+        val taskTranslator = CwlToolTranslator.apply(tool, sourceFile)
+        taskTranslator.apply(tool, sourceFile)
+      case _: Workflow => throw new NotImplementedError("Workflows are not implemented!")
+      case other => throw new NotImplementedError(s"Cwl type ${other.getClass} is not supported!")
+    }
+  }
+
+  def parseCommandOutputParameter(output: CommandOutputParameter): Parameter = {
+    val name = output.id.get.name.get
+    val dxType = CwlUtils.toIRType(output.types)
+    val attributes: Vector[ParameterAttribute] = CwlUtils.createParameterAttributes(output.doc, output.label)
+
+    Parameter(name, dxType, attributes=attributes)
+  }
+
+  def parseCommandInputParameter(input: CommandInputParameter): Parameter = {
+    val name = input.id.get.name.get
+    val dxType = CwlUtils.toIRType(input.types)
+    val defaultValue = Option(CwlUtils.toIRValue(input.default.getOrElse(CwlUtils.getDefaultCWLValue(input.types))))
+    val attributes: Vector[ParameterAttribute] = CwlUtils.createParameterAttributes(input.doc, input.label)
+
+    Parameter.apply(name.toString, dxType, defaultValue, attributes)
+  }
+
+  private case class CwlToolTranslator(tool: Process, sourceFile: Path) {
+
+    def apply(tool: CommandLineTool, sourceFile: Path): Application = {
```

Comment:
you don't need any arguments here - you already based them to the case class constructor.